docs/core-team-process.md

```diff
@@ -0,0 +1,77 @@
+
+:+1: _Advantages:_
+
+* Encourages high-touch, continuous conversation involving and relevant to the full team.
+* Resilient to time off and asynchronicity.
+* Minimizes the need to context switch up and down abstraction layers while working.
+
+:-1: _Disadvantages:_
+
+* Diminishes variety of work done by any individual developer, which could become boring.
+* Reduces the familiarity developed by any single developer to a single abstraction layer within the codebase.
+* Timing may become difficult. It's possible that one "seam" may take much more time to implement than the others, which could lead to a bottleneck.
+* Some efforts will not be decomposable into easily identified seams for division of labor.
+
+### 2. Pull request hierarchy
+
+The problem at hand is decomposed into a queue of relatively independent tasks to complete. A primary branch and a pull request are created to accumulate the full, shippable solution on full completion. Each developer creates an individual branch from the primary one and pushes commits as they work, opening a pull request that targets the primary branch as a base. Developers review one another's sub-pull requests with pull request reviews and coordinate merges to the primary until all tasks are complete, at which point the primary pull request is merged.
+
+> Example: developers A and B create and push a parent branch `a-b/big-feature` and open pull request 123 with an overall problem definition and a checklist of tasks to complete. Developer A creates branch `a/user-story-a` from `a-b/big-feature` and opens pull request 444 while developer B works on branch `b/user-story-b` and pull request 555. Developer A reviews and merges pull request 555 while developer B moves on to branch `b/user-story-c`, then developer B reviews and merges pull request 444. Developers A and B continuously calibrate the task list to represent the remaining work. Once the task list is complete, the primary pull request 123 is merged and the feature is shipped.
+
+:+1: _Advantages:_
+
+* Makes it less likely that one developer may block the others when their tasks take longer than expected.
+* More asynchronous-friendly.
+* Leaves a trail of documentation for each task.
+
+:-1: _Disadvantages:_
+
+* Decomposing tasks well is challenging.
```

Comment:
another disadvantage is having to coordinate when you're merging things.  If `user-story-c` finishes before `user-story-b`, and then you want to build a `user-story-d` that depends on `user-story-c`, that's going to be somewhat annoying.  (Maybe this is covered by "decomposing tasks well is challenging"?  ¯\_(ツ)_/¯

Reply:
Yeah, I think all of these approaches have some higher risk of merge conflicts. This one defers the resolution until it would be maximally painful though.